qkernel/src/interrupt/x86_64/mod.rs

```diff
@@ -425,11 +435,20 @@ bitflags! {
 
 #[no_mangle]
 pub extern "C" fn PageFaultHandler(ptRegs: &mut PtRegs, errorCode: u64) {
+    #[cfg(feature = "cc")]
+    let mut cr2: u64;
+    #[cfg(not(feature = "cc"))]
     let cr2: u64;
     unsafe { asm!("mov {0}, cr2", out(reg) cr2 ) };
     let cr3: u64;
     unsafe { asm!("mov {0}, cr3", out(reg) cr3 ) };
 
+    #[cfg(feature = "cc")]
+    if snp_active() {
+        let c_bit_mask = C_BIT_MASK.load(Ordering::Relaxed);
+        cr2 &= !c_bit_mask;
```

Comment:
snp is activated by setting a bit in cr2? interesting :thinking:

Reply:
Here is to remove the bit if it is set. Make it compatible with the normal handling logic.